test_grep_error is failing with newer git version

During my developement I found that there are two failig tests on `master` related to grep.
The root cause of this, there was a change in git upstream related to debug message:
`no pattern given`
It has been changed from:  `no pattern given.` to `no pattern given`

Url for the change:
https://github.com/git/git/commit/1a07e59c3e269418f3f5d186d166bf5ab5db6667
Search for the: builtin/grep.c file

Reproduction:
`$ python3 -m pytest tests/grep_test.py`

Test error:
```
file_config_files = auto_namedtuple(output_dir=local('/tmp/pytest-of-micek/pytest-6/test_grep_error_args0_0/output'), cfg=local('/tmp/pyte...ep_error_args0_0/2'), rev1='09b8d2e4a13de36711107a7814bed289708aee3c', rev2='75c5862c260933404547b4b001ba39eaca130454')
capfd = <_pytest.capture.CaptureFixture object at 0x7fc80abaed30>, args = ()

    @pytest.mark.parametrize('args', ((), ('--repos-with-matches',)))
    def test_grep_error(file_config_files, capfd, args):
        ret = main(('-C', str(file_config_files.cfg), *args))
        assert ret == 128
        out, err = capfd.readouterr()
        assert out == ''
>       assert err == 'fatal: no pattern given.\n'
E       AssertionError: assert equals failed
E         'fatal: no pattern given\n'      'fatal: no pattern given.\n'

args       = ()
capfd      = <_pytest.capture.CaptureFixture object at 0x7fc80abaed30>
err        = 'fatal: no pattern given\n'
file_config_files = auto_namedtuple(output_dir=local('/tmp/pytest-of-micek/pytest-6/test_grep_error_args0_0/output'), cfg=local('/tmp/pytest-of-micek/pytest-6/test_grep_error_args0_0/config.json'), repos_json=local('/tmp/pytest-of-micek/pytest-6/test_grep_error_args0_0/repos.json'), dir1=local('/tmp/pytest-of-micek/pytest-6/test_grep_error_args0_0/1'), dir2=local('/tmp/pytest-of-micek/pytest-6/test_grep_error_args0_0/2'), rev1='09b8d2e4a13de36711107a7814bed289708aee3c', rev2='75c5862c260933404547b4b001ba39eaca130454')
out        = ''
ret        = 128

tests/grep_test.py:125: AssertionError
>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>> entering PDB >>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>
> /home/micek/workspace/github-mitzkia/all-repos/tests/grep_test.py(125)test_grep_error()
-> assert err == 'fatal: no pattern given.\n'
(Pdb) q
Exit: Quitting debugger
```

my environment:
```
$ python3 --version --version
Python 3.6.7 (default, Oct 22 2018, 11:32:17) 
[GCC 8.2.0]
```
```
$ python3 -m pytest --version
This is pytest version 4.3.1, imported from /usr/local/lib/python3.6/dist-packages/pytest.py
setuptools registered plugins:
  pytest-xdist-1.26.1 at /usr/local/lib/python3.6/dist-packages/xdist/plugin.py
  pytest-xdist-1.26.1 at /usr/local/lib/python3.6/dist-packages/xdist/looponfail.py
  pytest-sugar-0.9.2 at /usr/local/lib/python3.6/dist-packages/pytest_sugar.py
  pytest-icdiff-0.0.4 at /usr/local/lib/python3.6/dist-packages/pytest_icdiff.py
  pytest-forked-1.0.1 at /usr/local/lib/python3.6/dist-packages/pytest_forked/__init__.py
  pytest-env-0.6.2 at /usr/local/lib/python3.6/dist-packages/pytest_env/plugin.py
  pytest-cov-2.6.1 at /usr/local/lib/python3.6/dist-packages/pytest_cov/plugin.py
  hypothesis-4.5.2 at /usr/local/lib/python3.6/dist-packages/hypothesis/extra/pytestplugin.py
```

```
$ git version 2.19.1
```

--- a/tests/grep_test.py
+++ b/tests/grep_test.py
@@ -122,4 +122,7 @@ def test_grep_error(file_config_files, capfd, args):
     assert ret == 128
     out, err = capfd.readouterr()
     assert out == ''
-    assert err == 'fatal: no pattern given.\n'
+    assert err in {
+        'fatal: no pattern given.\n',  # git < v2.19.0-rc0
+        'fatal: no pattern given\n',  # git >= v2.19.0-rc0
+    }

